pkg/distributor/errors.go

```diff
@@ -253,39 +253,41 @@ func (e circuitBreakerOpenError) Cause() mimirpb.ErrorCause {
 // Ensure that circuitBreakerOpenError implements Error.
 var _ Error = circuitBreakerOpenError{}
 
-// toGRPCError converts the given error into an appropriate gRPC error.
-func toGRPCError(pushErr error, serviceOverloadErrorEnabled bool) error {
+// toErrorWithGRPCStatus converts the given error into an appropriate gRPC error.
+func toErrorWithGRPCStatus(pushErr error, serviceOverloadErrorEnabled bool) error {
 	var (
 		distributorErr Error
 		errDetails     *mimirpb.ErrorDetails
 		errCode        = codes.Internal
 	)
 	if errors.As(pushErr, &distributorErr) {
 		errDetails = &mimirpb.ErrorDetails{Cause: distributorErr.Cause()}
-		switch distributorErr.Cause() {
-		case mimirpb.BAD_DATA:
-			errCode = codes.FailedPrecondition
-		case mimirpb.INGESTION_RATE_LIMITED:
-			errCode = codes.ResourceExhausted
-		case mimirpb.REQUEST_RATE_LIMITED:
-			if serviceOverloadErrorEnabled {
-				errCode = codes.Unavailable
-			} else {
-				errCode = codes.ResourceExhausted
-			}
-		case mimirpb.REPLICAS_DID_NOT_MATCH:
-			errCode = codes.AlreadyExists
-		case mimirpb.TOO_MANY_CLUSTERS:
-			errCode = codes.FailedPrecondition
-		case mimirpb.CIRCUIT_BREAKER_OPEN:
-			errCode = codes.Unavailable
-		case mimirpb.METHOD_NOT_ALLOWED:
-			errCode = codes.Unimplemented
-		}
+		errCode = toGRPCStatusCode(distributorErr.Cause(), serviceOverloadErrorEnabled)
 	}
 	return globalerror.WrapErrorWithGRPCStatus(pushErr, errCode, errDetails).Err()
 }
 
+func toGRPCStatusCode(errCause mimirpb.ErrorCause, serviceOverloadErrorEnabled bool) codes.Code {
+	switch errCause {
+	case mimirpb.BAD_DATA:
+		return codes.InvalidArgument
```

Comment:
Does the change from `codes.FailedPrecondition` to `codes.InvalidArgument` make any practical difference?

Reply:
No, this gRPC code is never used. It is returned by `distributor.Push()` (endpoint `/distributor.Distributor/Push`) which is not used.